blocks/bricks/recurrent.py

```diff
@@ -198,12 +200,22 @@ def scan_function(*args):
             outputs_info = (list(states_given.values()) +
                             [None] * (len(application.outputs) -
                                       len(application.states)))
+            if config.scan_strict:
+                # Find all shared variables
+                shared_vars = list(Selector(brick).get_params().values())
```

Comment:
I think you should leave the global configuration, but use it like I
suggested in my code snippet. It'd be nice if a user can speed things up
easily by just setting one flag, while maintaining fine-grained control
when needed.
On Mar 19, 2015 12:29 PM, "dmitriy-serdyuk" <notifications@github.com>
wrote:

> In blocks/bricks/recurrent.py
> <https://github.com/bartvm/blocks/pull/486#discussion_r26771158>:
>
> > @@ -198,12 +200,22 @@ def scan_function(*args):
> >              outputs_info = (list(states_given.values()) +
> >                              [None] * (len(application.outputs) -
> >                                        len(application.states)))
> > +            if config.scan_strict:
> > +                # Find all shared variables
> > +                shared_vars = list(Selector(brick).get_params().values())
>
> Right, it's better than introducing a new global configuration parameter.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bartvm/blocks/pull/486/files#r26771158>.
>